pkg/crd/markers/crd.go

```diff
@@ -48,6 +48,9 @@ var CRDMarkers = []*definitionWithHelp{
 
 	must(markers.MakeDefinition("kubebuilder:unservedversion", markers.DescribesType, UnservedVersion{})).
 		WithHelp(UnservedVersion{}.Help()),
+
+	must(markers.MakeDefinition("kubebuilder:deprecated", markers.DescribesType, DeprecatedVersion{})).
```

Comment:
should this be `deprecatedversion` to match with `unservedversion` and in case we ever get field-level deprecation, or do we think we can just use the same marker for field-level deprecation if we ever get it?

Reply:
it makes sense to keep them consistent, I changed it to `deprecatedversion`